### DISCUSSION TITLE
Changing UWP export options

### DISCUSSION THREAD

**Discussion by sweethoneycode** [godotengine/godot-proposals#5364] [Platforms]
I have been looking into GADOT with the ability to export to UWP and it is very broken. It attempts to export to .appx format and tends to be empty. This is more or less due to the package manifest not formatted correctly. I am wondering if this is not the best approach. 

Before trying GADOT I have been using Unity for over 10 years. It's UWP export takes a different approach. It packages a Visual Studio Solution for UWP and it does this with C++. It essentially packages it to have the current manifests and directories needed to build the correct files for Windows UWP format which is still needed for the Windows Store. (My games have high installs on this store).

Through Visual Studio the APP can be associated to the correct store listing or create a new one and more. Updates can be done to the app manifest without making an error in the XML. In short the heavy lifting of meeting the requirements for the Windows Store is handed off to Visual Studio which is available on Windows or MAC.

I understand this may not be a good option for Linux users, but with my experience with using Unity it just works overall.

I don't know what it would take to build such a solution for GADOT. However GADOT is easier to develop in and I would be able to transition to this fully if it were available.

Thanks!

**Comment by Calinou** [comment]
The UWP export gets very little maintenance, as Microsoft themselves is no essentially no longer caring about UWP. There are very few contributors who are knowledgable with UWP. Even then, most of them would rather spend their time improving the Win32 platform port which has far more users and has a brighter future.

PS: It's "Godot", not "GADOT" :slightly_smiling_face:

**Reply by sweethoneycode** [reply]
Sorry for the misspelling. I do understand as I’ve seen that through the forums. However, the Windows store still requires either: .msix, .appx or appxupload. I’ve seen games there, is there another way since the team isn’t vested into the UWP?

thanks

**Reply by Calinou** [reply]
> I’ve seen games there, is there another way since the team isn’t vested into the UWP?

See comments on https://github.com/godotengine/godot/issues/17985.

**Reply by sweethoneycode** [reply]
Thanks! I would like to help make a help doc page encouraging using those tools instead for the windows store and maybe the UWP should have a note that it's not fully supported.
